src/main/plugins/org.dita.xhtml/xsl/xslhtml/troubleshootingdisplay.xsl

```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- This file is part of the DITA Open Toolkit project hosted on 
+     Sourceforge.net. See the accompanying license.txt file for 
+     applicable licenses.-->
+<!-- 20170503 SCH: Add support for troubleshooting elements. -->
+
+<xsl:stylesheet version="2.0"
+     xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+     xmlns:related-links="http://dita-ot.sourceforge.net/ns/200709/related-links"
+     xmlns:dita2html="http://dita-ot.sourceforge.net/ns/200801/dita2html"
+     xmlns:ditamsg="http://dita-ot.sourceforge.net/ns/200704/ditamsg"
+     xmlns:xs="http://www.w3.org/2001/XMLSchema"
+     exclude-result-prefixes="related-links dita2html ditamsg xs">
+
+<!-- Determines whether to generate titles for task sections. Values are YES and NO. -->
+<xsl:param name="GENERATE-TASK-LABELS" select="'NO'"/>
+
+<!-- == TASK UNIQUE SUBSTRUCTURES == -->
+
+  <xsl:template match="*[contains(@class,' troubleshooting/troublebody ')]" name="topic.troubleshooting">
+<div>
+  <xsl:call-template name="commonattributes"/>
+  <xsl:call-template name="setidaname"/>
+  <xsl:apply-templates select="*[contains(@class,' ditaot-d/ditaval-startprop ')]" mode="out-of-line"/>
+  <!-- here, you can generate a toc based on what's a child of body -->
+  <!--xsl:call-template name="gen-sect-ptoc"/--><!-- Works; not always wanted, though; could add a param to enable it.-->
+
+  <!-- Added for DITA 1.1 "Shortdesc proposal" -->
+  <!-- get the abstract para -->
+  <xsl:apply-templates select="preceding-sibling::*[contains(@class,' topic/abstract ')]" mode="outofline"/>
+
+  <!-- get the short descr para -->
+  <xsl:apply-templates select="preceding-sibling::*[contains(@class,' topic/shortdesc ')]" mode="outofline"/>
+
+  <xsl:apply-templates/>
+  <xsl:apply-templates select="*[contains(@class,' ditaot-d/ditaval-endprop ')]" mode="out-of-line"/>
+</div><xsl:value-of select="$newline"/>
+</xsl:template>
+
+<xsl:template match="*[contains(@class,' task/steptroubleshooting ')]" name="topic.task.steptroubleshooting">
```

Comment:
As with the comments on FO -- task elements should be handled in the task module.